js/services/auth.js

```diff
@@ -21,13 +23,58 @@ const signUpUser = (signUpDetails) => {
       .catch((error) => {
         const signUpResult = {
           uId: null,
+          signedIn: { signUp: false, signIn: false },
+          userName: {},
           message: (`Sorry, sign-up was unsuccessful. The following error has occured: ${error.code}, ${error.message}`),
         };
 
         return signUpResult;
       });
 };
 
+const signInUser = (signInDetails) => {
+  return firebase.auth()
+    .signInWithEmailAndPassword(signInDetails.email, signInDetails.password)
+    .then(() => {
+      const uId = firebase.auth().currentUser.uid;
+
+      // get userDetails from firebase and return signInResult to thunk
+      return userDetails.get(uId)
+        .then((result) => {
+          const signInResult = {
+            uId,
+            signedIn: { signUp: false, signIn: true },
+            userName: result,
+            message: (`Sign-in successful, welcome back to Pot Lucky ${result.firstName}`),
+          };
+
+          return signInResult;
+        });
+    })
```

Comment:
You're nesting promise chains, which is unnecessary. Instead just `return userDetails.get(uId)` and then and the subsequent code in the next `.then`.